src/RelayServer.ts

```diff
@@ -486,12 +486,8 @@ export class RelayServer extends EventEmitter {
     isSponsoredTx(req: RelayRequest | DeployRequest): boolean {
         const { disableSponsoredTx, sponsoredDestinations } = this.config;
 
-        if (!disableSponsoredTx) {
-            return true;
-        }
-
         if (
-            sponsoredDestinations &&
+            !disableSponsoredTx ||
```

Comment:
I don't think we should check for this here. The function name is `isSponsoredTx`, so IMHO here we should check if the transaction is sponsored and not if the server is configured to accept sponsored transaction.

Reply:
maybe changing the name to isTxSponsored, or do you think that we should have different functions for each validation?

Reply:
updated